ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -4007,9 +4007,17 @@ public boolean getPartitionsByExpr(Table tbl, ExprNodeGenericFuncDesc expr, Hive
       if (AcidUtils.isTransactionalTable(tbl)) {
         validWriteIdList = getValidWriteIdList(tbl.getDbName(), tbl.getTableName());
       }
-      boolean hasUnknownParts = getMSC().listPartitionsSpecByExpr(tbl.getDbName(),
-          tbl.getTableName(), exprBytes, defaultPartitionName, (short)-1, msParts,
-          validWriteIdList != null ? validWriteIdList.toString() : null);
+
+      PartitionsByExprRequest req = new PartitionsByExprRequest(tbl.getDbName(), tbl.getTableName(),
```

Comment:
It will be good to encapsulate creation of `PartitionsByExprRequest` is a separate method.